.github/workflows/push.yml

```diff
@@ -123,6 +123,71 @@ jobs:
           name: fastfetch-bsd
           path: ./fastfetch-*.*
 
+  windows:
+    name: Windows
+    runs-on: windows-latest
+    permissions:
+      security-events: write
+      contents: read
+    outputs:
+      ffversion: ${{ steps.ffversion.outputs.ffversion }}
+    defaults:
+      run:
+        shell: msys2 {0}
+    steps:
+      - name: checkout repository
+        uses: actions/checkout@v3
+
+      - name: setup-msys2
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: MSYS
+          update: true
+          install: git base-devel cmake gcc mingw-w64-clang-x86_64-cjson mingw-w64-clang-x86_64-vulkan-loader mingw-w64-clang-x86_64-opencl-icd
+
+      - name: print msys version
+        run: uname -a
+
+      # https://github.com/msys2/MINGW-packages/issues/13524#event-7555720785
+      - name: create OpenCL.pc
+        run: |
+          cat > /clang64/lib/pkgconfig/OpenCL.pc << EOF
+          prefix=/clang64
+          exec_prefix=${prefix}
+          libdir=${exec_prefix}/lib
+          includedir=${prefix}/include
+
+          Name: OpenCL
+          Description: Open Computing Language generic Installable Client Driver Loader
+          Version: 2022.09.30-1
+          Libs: -L${libdir} -lOpenCL.dll
+          Cflags: -I${includedir}
+          EOF
+
+      - name: configure project
+        run: env PKG_CONFIG_PATH=/clang64/lib/pkgconfig/:$PKG_CONFIG_PATH cmake -DSET_TWEAK=Off -DBUILD_TESTS=On .
+
+      - name: build project
+        run: cmake --build . --target fastfetch --target flashfetch # Makes no sense to install exes to /usr/bin for Windows
+
+      - name: copy necessary dlls
+        run: cp /usr/bin/msys-2.0.dll /clang64/bin/*.dll .
+
+      # Crashes on start for some reason, but it provides binaries at least. Needs investigation.
+      # - name: run fastfetch
+      #   run: ./fastfetch --recache --disable-linewrap false --hide-cursor false --show-errors true --load-config presets/all
+
+      # - name: run flashfetch
+      #   run: ./flashfetch
+
+      - name: upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: fastfetch-windows
+          path: |
+            ./*.dll
+            ./*.exe
```

Comment:
I think we should update CPack to build an windows installer and provide that instead of raw exe + dll.

Reply:
I'm not familier with CPack, but I don't think fastfatch should be *installed* in `Program Files` or somewhere because it's not where CUI programs should be and it's not in global $PATH.

Same case of macOS, we should not provide .dmg because files in .dmg usually means that they should be dropped into `/Applications`. CUI tools like fastfetch should always be installed with package manager or just be put into somewhere in user's home folder.

Reply:
I am not familiar with the install process on those platforms, but this means we should remove the dmg.

Reply:
Can we at least zip the exe and the dlls together?

Reply:
They will be zipped together by `actions/upload-artifact@v3`

Note Windows part doesn't go to release action (yet)